src/routes/index.ts

```diff
@@ -0,0 +1,67 @@
+import { Application } from "express";
+import { errorHandler } from "../middlewares/errorHandler.middleware";
+import { notFoundHandler } from "../middlewares/errorHandler.middleware";
+
+class AppRouter {
+  constructor(private app: Application) {}
+  init() {
+       /**
+     * @swagger
+     * /health-check:
+     *   get:
+     *     summary: Health Check endpoint
+     *     description: Returns a JSON response indicating the server is running.
+     *     responses:
+     *       200:
+     *         description: Server is running
+     *         content:
+     *           application/json:
+     *             schema:
+     *               type: object
+     *               properties:
+     *                 message:
+     *                   type: string
+     *                   example: Server is running
+     *                 app:
+     *                   type: string
+     *                   example: Health
+     *       404:
+     *         description: Resource not found
+     *       500:
+     *         description: Internal Server Error
+     */
+    this.app.get('/health-check', (req, res) => {
+      res.status(200).json({message: 'Server is running', app: 'Health'})
+    });
+    this.app.use('/health-check', [notFoundHandler, errorHandler] ),
```

Comment:
You can set an error handler at the end of the init method once and it should handle all of errors from routes.
http://expressjs.com/en/guide/error-handling.html#error-handling

Reply:
Simplified the middlewares' setup for the route